runners/core-java/src/main/java/org/apache/beam/runners/core/SimpleDoFnRunner.java

```diff
@@ -842,6 +842,32 @@ public void setForNowPlus(Duration durationFromNow) {
       setUnderlyingTimer(target);
     }
 
+    @Override
+    public void setForNowAlign(Duration size, Duration durationFromNow) {
+      Instant now = getCurrentTime();
+      long millisSinceStart = now.plus(durationFromNow).getMillis() % size.getMillis();
+      Instant target = millisSinceStart == 0 ? now : now.plus(size).minus(millisSinceStart);
+      target = minTargetAndGcTime(target);
+      setUnderlyingTimer(target);
+    }
+
+    /**
+     * For event time timers the target time should be prior to window GC time. So it return
+     * min(time to set, GC Time of window).
+     */
+    private Instant minTargetAndGcTime(Instant target) {
+      if (TimeDomain.EVENT_TIME.equals(spec.getTimeDomain())) {
```

Comment:
Readability nit (not required): switch the if/else so the short one comes first. You can also eliminate a level of nesting if you just let the `return` end it.